### PR TITLE
Selection to Point and some dropdowns

### PR DESCRIPTION
**Pull request by helblazer811**
I implemented selection to point for cylinders, spheres, lines, and arrows. I am not sure if this is all of the things that could use it. I also added drop downs for a couple things and refactored a bit.

### CODE REVIEW

**Comment by helblazer811** [comment]
This should resolve issue #315

File: 3Dmol/glviewer.js [hunk 1]
```diff
@@ -1820,6 +1829,10 @@ $3Dmol.GLViewer = (function() {
          */
         this.addSphere = function(spec) {
             spec = spec || {};
+            if(!spec.center.hasOwnProperty("x") || !spec.center.hasOwnProperty("y") || !spec.center.hasOwnProperty("z")){
```
Comment (dkoes) [inline 3Dmol/glviewer.js]: I would factor this out as well into the helper function.  And try to make it robust to failure (what if getAtomsFromSel returns nothing? - should provide reasonable defaults).

File: tests/auto/tests/test92.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+                viewer.addCylinder({start:{},
+                                  end:{x:20.0,y:0.0,z:0.0},
+                                  radius:1.0,
+                                  color:'black',
+                                  fromCap:false,
+                                  toCap:false});
+                                   viewer.addLine({dashed:true,start:{x:0,y:0,z:0},end:{}});
+
+                viewer.addArrow({
+                      start: {},
```
Comment (dkoes) [inline tests/auto/tests/test92.js]: let's have a selection more complex than everything, e.g. `{resi: 31}`

File: tests/auto/tests/test92.js [hunk 3]
```diff
@@ -9,7 +8,7 @@
                                   color:'black',
                                   fromCap:false,
                                   toCap:false});
-                                   viewer.addLine({dashed:true,start:{x:0,y:0,z:0},end:{}});
+                viewer.addLine({dashed:true,start:{x:0,y:0,z:0},end:{}});
 
                 viewer.addArrow({
                       start: {},
```
Comment (dkoes) [inline tests/auto/tests/test92.js]: still looking for a more complex example..

**Pull request merged**